src/Tinfoil/Data/Token.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DeriveGeneric #-}
+module Tinfoil.Data.Token(
+    SecureToken (..)
+  , secureTokenLength
+  ) where
+
+import           Control.DeepSeq.Generics (genericRnf)
+
+import           GHC.Generics (Generic)
+
+import           P
+
+-- | Cryptographically secure identifier,
+-- for things like API keys or password reset tokens.
+-- Provided as a 48 * 4 length string with base16 encoding
```

Comment:
This comment doesn't match `secureTokenLength` below. Final hex-encoded token length will be 32 bytes/256 bits for a raw token size of 16 bytes/128 bits.